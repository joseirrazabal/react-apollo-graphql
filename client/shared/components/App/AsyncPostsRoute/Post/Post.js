/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { shape, bool, string, number } from 'prop-types';
import { gql, graphql } from 'react-apollo';
import Helmet from 'react-helmet';
// import SINGLE_POST_QUERY from './singlePost.graphql';

export class Post extends Component {
  render() {
    const { data: { loading, singlePost } } = this.props;
    if (loading) {
      return <h1>LOADING...</h1>;
    }
    const { title, body, userId } = singlePost;
    return (
      <div>
        <Helmet title={`Posts - ${title}`} />
        <div>
          <h1>
            {title}
          </h1>
          <p>
            {body}
          </p>
          <hr />
          <span>
            By User {userId}
          </span>
        </div>
      </div>
    );
  }
}
export const SINGLE_POST_QUERY = gql`
  query singlePost($id: Int!) {
    singlePost(id: $id) {
      id
      title
      userId
      body
    }
  }
`;
export default graphql(SINGLE_POST_QUERY, {
  options: props => ({
    variables: {
      id: props.match.params.id,
    },
  }),
})(Post);

Post.propTypes = {
  data: shape({
    loading: bool.isRequired,
    singlePost: shape({
      title: string,
      body: string,
      userId: number,
    }),
  }).isRequired,
};
