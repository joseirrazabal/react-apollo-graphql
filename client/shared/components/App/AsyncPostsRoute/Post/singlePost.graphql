query singlePost($id: Int!) {
  singlePost(id: $id) {
    id
    title
    userId
    body
  }
}
