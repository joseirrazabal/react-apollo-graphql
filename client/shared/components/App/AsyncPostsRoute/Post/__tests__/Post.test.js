/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import { shallow } from 'enzyme';

import { Post } from '../Post';

const data = {
  loading: false,
  singlePost: {
    id: 1,
    title: 'hi',
    body: 'boo',
    userId: '3',
  },
};

describe('<Post />', () => {
  test('renders header', () => {
    const wrapper = shallow(<Post data={data} />);
    expect(wrapper.find('h1').length).toBe(1);
  });
  test('renders the snapshot', () => {
    const wrapper = shallow(<Post data={data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
