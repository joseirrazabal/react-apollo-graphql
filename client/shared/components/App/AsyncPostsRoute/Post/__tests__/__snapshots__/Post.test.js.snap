// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> renders the snapshot 1`] = `
<div>
  <HelmetWrapper
    encodeSpecialCharacters={true}
    title="Posts - hi"
  />
  <div>
    <h1>
      hi
    </h1>
    <p>
      boo
    </p>
    <hr />
    <span>
      By User 
      3
    </span>
  </div>
</div>
`;
